desc/compute/__init__.py

```diff
@@ -94,3 +94,30 @@ def _build_data_index():
 
 
 _build_data_index()
+
+
+def set_tier(name, p):
+    """Determine how deep in the dependency tree a given name is.
+
+    tier of 0 means no dependencies on other data,
+    tier of 1 means it depends on only tier 0 stuff,
+    tier of 2 means it depends on tier 0 and tier 1, etc etc.
+
+    Designed such that if you compute things in the order determined by tiers,
+    all dependencies will always be computed in the correct order.
+    """
+    if "tier" in data_index[p][name]:
+        return
+    if len(data_index[p][name]["full_with_axis_dependencies"]["data"]) == 0:
+        data_index[p][name]["tier"] = 0
+    else:
+        thistier = 0
```

Comment:
I don't think it will matter, but technically this should be `thistier = 1` since being in this else statement means that it has a finite amount of dependencies and therefore must be at least tier 1 or higher.

Reply:
Also something like `dep_tier` for "dependency tier" might be a better variable name

Reply:
This is correct as written; setting `thistier` to anything greater than 0 would be incorrect. We add +1 in line 118 to max dependency tier.